tests/functional/SaveQueryValidateCest.php

```diff
@@ -0,0 +1,125 @@
+// "query ($num: ID!) { posts { edges { node { title } } } }"
+
+class SaveQueryValidateCest {
+	public function _before( FunctionalTester $I ) {
+		// Make sure that is gone.
+		$I->dontHavePostInDatabase(['post_title' => 'Hello world!']);
+
+		// clean up and persisted queries terms in the taxonomy
+		$I->dontHavePostInDatabase( [ 'post_type' => 'graphql_document' ] );
+		$I->dontHaveTermInDatabase( [ 'taxonomy' => 'graphql_query_alias'] );
+
+		$I->dontHaveOptionInDatabase( 'graphql_persisted_queries_section'  );
+	}
+
+	public function _after( FunctionalTester $I ) {
+		$I->dontHaveOptionInDatabase( 'graphql_persisted_queries_section'  );
+
+		// Clean up any saved query documents
+		$I->dontHavePostInDatabase( [ 'post_type' => 'graphql_document' ] );
+		$I->dontHaveTermInDatabase( [ 'taxonomy' => 'graphql_query_alias'] );
+	}
+
+	public function saveQueryWithArgumentMissingVariableTest( FunctionalTester $I ) {
+		$query = 'query { post( id: $num, idType: DATABASE_ID) { title } }';
+		$query_alias = 'test-save-query-alias';
+		$I->dontSeeTermInDatabase( [ 'name' => 'graphql_query_alias' ] );
+		$I->sendPost('graphql', [
+			'query' => $query,
+			'queryId' => $query_alias
+		] );
+		$I->dontSeePostInDatabase( [
+			'post_type'    => 'graphql_document',
+			'post_status'  => 'publish',
+		] );
+		$I->seeResponseContainsJson([
+			'errors' => [
+				'message' => 'Validation Error: Variable "$num" is not defined.'
+			]
+		]);
+	}
+
+	public function saveQueryMissingArgumentWithVariableTest( FunctionalTester $I ) {
+		$query = 'query ($num: ID!) { post( id: "1", idType: DATABASE_ID ) { title } }';
+		$query_alias = 'test-save-query-alias';
+		$I->dontSeeTermInDatabase( [ 'name' => 'graphql_query_alias' ] );
+		$I->sendPost('graphql', [
+			'query' => $query,
+			'queryId' => $query_alias
+		] );
+		$I->dontSeePostInDatabase( [
+			'post_type'    => 'graphql_document',
+			'post_status'  => 'publish',
+		] );
+		$I->seeResponseContainsJson([
+			'errors' => [
+				'message' => 'Validation Error: Argument "$id" should be a variable.'
+			]
+		]);
+	}
+
+	public function saveQueryWithArgumentWithVariableTest( FunctionalTester $I ) {
+		$query = 'query ($num: ID!) { post( id: $num, idType: DATABASE_ID) { title } }';
+		$query_alias = 'test-save-query-alias';
+
+		$post_id = $I->havePostInDatabase(['post_title' => 'Hello world!']);
+
+		$I->dontSeeTermInDatabase( [ 'name' => 'graphql_query_alias' ] );
+		$I->sendPost('graphql', [
+			'query' => $query,
+			'queryId' => $query_alias,
+			'variables' => [ "num" => $post_id ],
+		] );
+		$I->seeResponseContainsJson([
+			'data' => [
+				'post' => [
+					'title' => 'Hello world!',
+				]
+			]
+		]);
+		$I->seeTermInDatabase( [ 'name' => $query_alias ] );
+		$I->seePostInDatabase( [
+			'post_type'    => 'graphql_document',
+			'post_status'  => 'publish',
+			'post_content' => "query (\$num: ID!) {\n  post(id: \$num, idType: DATABASE_ID) {\n    title\n  }\n}\n",
```

Comment:
This query _should_ fail validation because the `idType` could be concatenated still, leading to duplicate query documents being stored for different possible values. 

i.e.

```js
const idTye = 'DATABASE_ID';
const query = `query ($num: ID!) { post( id: $num, idType: ${idTye}) { title } }`
```

or: 

```js
const idTye = 'SLUG';
const query = `query ($num: ID!) { post( id: $num, idType: ${idTye}) { title } }`
```

By using string interpolation / concatenation, we'd be able to have multiple versions of the same query, which is the situation we're trying to prevent. 

Ideally we'd be guiding the user to: 

```js
const query = `query ($num: ID! $idType: PostIdType) { post( id: $num, idType: $idType }) { title } }`
```

Where here, each argument on the field level corresponds with a variable defined at the operation level.